src/marqo/tensor_search/throttling/check_and_increment.lua

```diff
@@ -0,0 +1,36 @@
+local set_key = KEYS[1]
+
+local thread_name = ARGV[1]
+local thread_limit = tonumber(ARGV[2])
+local expire_time = tonumber(ARGV[3])
+
+-- Get expiry time
+local now = redis.call('time')[1]
+local expiry_threshold = now - expire_time
+
+-- Expire items in sorted set past the threshhold
+redis.call('zremrangebyscore', set_key, '-inf', expiry_threshold)
+
+-- Current count is items in sorted set
+local current_thread_count = redis.call('zcard', set_key)
+
+if current_thread_count + 1 > thread_limit then
+    -- If over concurrency limit, return 0 for failure
+    return 0

```

Comment:
nit: typically 0 response means OK, any other number is error

Reply:
oh my bad. will change to -1